### PR TITLE
Refactor installPackage.m2, part 1

### PR DESCRIPTION
**Pull request by mahrud**
- [x] delete indexHtml and remove indexHtml.m2
- [x] move runFile and runString to run.m2
- [x] move sampleInitFile and readmeFile to system.m2
- [x] move TEST, sourceFileStamp, and check to testing.m2
- [x] viewHelp, about, apropos, and examples to help.m2
- [x] simplify html HEAD tag generation with new defaultHEAD() in html.m2
- [x] move URL to hypertext.m2
- [x] move showHtml and show for Hypertext and URL to html.m2 and showTex to expressions.m2
- [x] move show method declaration from engine.m2 to methods.m2

Changes for the next PR:
- [ ] break apart installPackage method itself
- [ ] simplify makePackageIndex method
- [ ] add markdown option to both
- [ ] move help method to help.m2
- [ ] move runnable(s) and browser() to system.m2?
- [ ] move show URL to somewhere more appropriate?

### CODE REVIEW

**Comment by DanGrayson** [comment]
Thank you for the contribution.

Could you document the removal of `showHtml` by adding an entry to the documentation node "changes made for the next release" in a list entitled "functionality removed", along with your workaround?

`showTex` might still be useful.

**Comment by mahrud** [comment]
> Could you document the removal of `showHtml` by adding an entry to the documentation node "changes made for the next release" in a list entitled "functionality removed", along with your workaround?

I had a call with Paul and had a change of heart about this. I restored showHtml in html.m2 and will look at this again later, potentially along with other changes to the "show" methods.

**Comment by mahrud** [comment]
@DanGrayson do you have any suggestions for how to prevent something like `show URL "http://google.com"` from printing browser output to Macaulay2 output? Maybe we need to add something like:
```c
int nullfd = open("/dev/null",0);
dup2(nullfd, 1);
dup2(nullfd, 2);
close(nullfd);
execvp(...);
```
In here:
https://github.com/Macaulay2/M2/blob/9e3c064aa61659f1052026c042aacbfea94f42c3/M2/Macaulay2/d/scclib.c#L98-L107

**Comment by DanGrayson** [comment]
"exec" is a system call, and "system_exec" provides low level access to it,
so it should behave as "exec" does
and not redirect stdout and stderr to /dev/null -- opening a browser is not
the only potential application of "exec".

If the browser output that annoys you is on stdout, you might consider
something like
  run "browser url >/dev/null &"
, but messages sent to stderr are error messages and should be seen by the
user.

Doing it that way would defeat the way I used setGroupId to put the browser
into a new process group.  Presumably
I wanted interrupts from the terminal (CTRL-c) to go just to M2 and not to
the browser.  It would be sad if
interrupting a computation caused the user's open browser to terminate.

**Comment by mahrud** [comment]
@DanGrayson pushed a commit that solves the issue you opened about example tables as well as an older issue. This PR is ready.

**Comment by DanGrayson** [comment]
If I use your code to install `FirstPackage`, then many superfluous blank lines now appear in the help:
Screen Shot 2020-06-23 at 1 13 12 PM

**Comment by DanGrayson** [comment]
Authors' email addresses now appear with "mailto:" prepended, which is not right:
Screen Shot 2020-06-23 at 1 29 08 PM

**Comment by DanGrayson** [comment]
Also, the email addresses now appear twice:
Screen Shot 2020-06-23 at 1 29 08 PM

**Comment by mahrud** [comment]
Yes, I fixed the email thing in another branch but forgot about it when I was rebasing the changes for this PR. I'll fix it.

Regarding the extra spaces, I thought it made it more readable, doesn't it? I think I can fix it if you don't like it.

**Comment by DanGrayson** [comment]
Vertical space is at a premium in terminals, so I prefer not to have extra blank lines.

**Comment by mahrud** [comment]
> Vertical space is at a premium in terminals, so I prefer not to have extra blank lines.

Both issues are fixed now, by the way.

File: M2/Macaulay2/m2/help.m2 [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            if packagesSeen#?pkgname then continue else packagesSeen#pkgname = 1;
+            dbname := pkg#"doc db file name";
+            dbkeys := keys fetchDocKeys pkg;
+            db := if o.Body then openDatabase dbname;
+            keyList := select(dbkeys, matchfun_re db);
+            if o.Body then close db;
+            apply(keyList, key -> pkgname | "::" | key))))
+
+-- TODO: why doesn't help ZZ work?
+help#0 ZZ := i -> (
```
Comment (DanGrayson) [inline M2/Macaulay2/m2/help.m2:104]: It does work:
```m2

i2 : about resolution

o2 = {0 => ChainComplexExtras::resolution(ChainComplex)               }
     {1 => Macaulay2Doc::resolution                                   }
     {2 => Macaulay2Doc::resolution(..., DegreeLimit => ...)          }
     {3 => Macaulay2Doc::resolution(..., FastNonminimal => ...)       }
     {4 => Macaulay2Doc::resolution(..., HardDegreeLimit => ...)      }
     {5 => Macaulay2Doc::resolution(..., LengthLimit => ...)          }
     {6 => Macaulay2Doc::resolution(..., PairLimit => ...)            }
     {7 => Macaulay2Doc::resolution(..., SortStrategy => ...)         }
     {8 => Macaulay2Doc::resolution(..., StopBeforeComputation => ...)}
     {9 => Macaulay2Doc::resolution(..., Strategy => ...)             }
     {10 => Macaulay2Doc::resolution(..., SyzygyLimit => ...)         }
     {11 => Macaulay2Doc::resolution(Ideal)                           }
     {12 => Macaulay2Doc::resolution(Matrix)                          }
     {13 => Macaulay2Doc::resolution(Module)                          }
     {14 => Macaulay2Doc::resolution(MonomialIdeal)                   }
     {15 => NCAlgebra::resolution(NCMatrix)                           }

o2 : NumberedVerticalList

i3 : help 11

o3 = Macaulay2Doc::resolution(Ideal) -- compute a projective resolution of (the quotient ring corresponding to) an ideal
     *******************************************************************************************************************

     +-----------------------------------------+
     |i1 : R = ZZ[a..d]                        |
     |                                         |
     |o1 = R                                   |
     |                                         |
     |o1 : PolynomialRing                      |
     +-----------------------------------------+
     |i2 : I = ideal(a,b,c,d)                  |
     |                                         |
     |o2 = ideal (a, b, c, d)                  |
     |                                         |
     |o2 : Ideal of R                          |
     +-----------------------------------------+
     |i3 : C = res I                           |
     |                                         |
     |      1      4      6      4      1      |
     |o3 = R  <-- R  <-- R  <-- R  <-- R  <-- 0|
     |                                         |
     |     0      1      2      3      4      5|
     |                                         |
     |o3 : ChainComplex                        |
     +-----------------------------------------+
     |i4 : C_2                                 |
     |                                         |
     |      6                                  |
     |o4 = R                                   |
     |                                         |
     |o4 : R-module, free, degrees {6:2}       |
     +-----------------------------------------+
     |i5 : C.dd_2                              |
     |                                         |
     |o5 = {1} | -b 0  -c 0  0  -d |           |
     |     {1} | a  -c 0  0  -d 0  |           |
     |     {1} | 0  b  a  -d 0  0  |           |
     |     {1} | 0  0  0  c  b  a  |           |
     |                                         |
     |             4       6                   |
     |o5 : Matrix R  <--- R                    |
     +-----------------------------------------+

     See also
     ========

       * "ChainComplex _ ZZ" -- component
       * "dd" -- differential in a chain complex
       * "resolution" -- projective resolution
       * "ideal" -- make an ideal

o3 : DIV
```

Comment (mahrud) [reply M2/Macaulay2/m2/help.m2:104]: I meant why doesn't `help ZZ := i -> ...` work?

Comment (DanGrayson) [reply M2/Macaulay2/m2/help.m2:104]: Oh, because `help` is actually a "Command", not a method function.

**Comment by mahrud** [comment]
Why do we have Commands?

**Comment by DanGrayson** [comment]
So typing "help" causes something to happen.  Compare:
```
Macaulay2, version 1.15
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases,
               PrimaryDecomposition, ReesAlgebra, TangentCone, Truncations

i1 : sin

o1 = sin

o1 : CompiledFunction
```

**Comment by DanGrayson** [comment]
... to this:
```

i2 : help

o2 = initial help
     ************

     Welcome to Macaulay2


     Try entering '2+2' at your next input prompt, which begins with i.  The two output
     prompts begin with o.  The first one, with the equal sign, '=', gives the value
     computed from your input, and the second one, with the colon, ':', tells what type of
     thing the value is.


     Type one of these commands to get started reading the documentation:
       * copyright                         -- the copyright
       * help "Macaulay2"                  -- top node of the documentation.
       * help "reading the documentation"  -- 
       * help "getting started"            -- 
       * help "a first Macaulay2 session"  -- 
       * help x                            -- show documentation for x
       * help about x                      -- show documentation about x
       * help about (x,Body=>true)         -- show documentation mentioning x
       * ? f                               -- display brief documentation for a function f
       * printWidth = 80                   -- set print width to 80 characters
       * viewHelp                          -- view documentation in a browser
       * viewHelp x                        -- view documentation on x in browser
     To read the documentation in info form, in case you happen to be running Macaulay2 in a
     terminal window, replace "help" by "infoHelp" in any of the commands above.

o2 : DIV
```

**Pull request merged**